stock_picking_consolidation_priority/models/stock_move.py

```diff
@@ -121,4 +121,19 @@ def _consolidate_priority(self):
                 [[("id", "in", move_ids)], self._consolidate_priority_domain()]
             )
         )
-        moves.picking_id.write(self._consolidate_priority_values())
+        for move in moves:
+            # Flag the move as urgent.
+            move.write(self._consolidate_priority_values())
+        for picking in moves.picking_id:
+            # Flag the picking as urgent if it only contains urgent moves.
+            picking_moves = picking.move_lines.filtered(
+                lambda m: m.state not in ("cancel", "done")
+            )
+            if (
+                all(
+                    m.priority == self._consolidate_priority_value
+                    for m in picking_moves
+                )
+                and picking.priority is not self._consolidate_priority_value
+            ):
```

Comment:
I'm pretty sure we can do all that in a single SQL query.
```sql
SELECT sp.id, ALL(sm.priority = {consolidate_value}) as has_consolidate_value, 
FROM stock_move sm
JOIN stock_picking sp ON sm.picking_id = sp.id
WHERE id in {move.picking_id}
AND sp.priority != {consolidate_value}
GROUP BY sm.picking_id
HAVING has_consolidate_value;
```
haven't tested though